index.bs

```diff
@@ -3701,6 +3727,792 @@ opened</dfn> steps given |window|, |type| and |message|.
+
+    1. For each |byte| in |value bytes|:
+
+      1. Append the [=byte/value=] of |byte| to |binary value|.
+
+1. Let |domain| be |stored cookie|'s domain field.
+
+1. Let |path| be |stored cookie|'s path field.
+
+1. Let |expires| be |stored cookie|'s expires field represented as a unix
+   timestamp, if set, or null otherwise.
+
+1. Let |size| be the byte length of the result of serializing |stored cookie|
+   as it would be represented in a <code>Cookie</code> header.
+
+1. Let |http only| be true if |stored cookie|'s http-only-flag is true, or false
+   otherwise.
+
+1. Let |secure| be true if |stored cookie|'s secure-only-flag is true, or false
+   otherwise.
+
+1. Let |same site| be "<code>none</code>" if |stored cookie|'s samesite-flag is
+   "<code>lax</code>" if it is "<code>Lax</code>, or "<code>strict</code>" if it
+   is "<code>Strict</code>".
+
+1. Return a map matching the <code>network.Cookie</code> production,
+   with the <code>name</code> field set to |name|, the <code>value</code> field
+   set to |value| if it's not null or omitted otherwise, the
+   <code>binaryValue</code> field set to |value bytes| if it's not null or
+   omitted otherwise, the <code>domain</code> field set to |domain|, the
+   <code>path</code> field set to |path|, the <code>expires</code> field set to
+   |expires| if it's not null, or omitted otherwise, the <code>size</code> field
+   set to |size|, the <code>httpOnly</code> field set to |http only|, the
+   <code>secure</code> field set to |secure|, and the <code>sameSite</code>
+   field set to |same site|.
+
+</div>
+
+#### The network.FetchTimingInfo type #### {#type-network-FetchTimingInfo}
+
+[=Remote end definition=] and [=local end definition=]
+
+<pre class="cddl local-cddl">
+network.FetchTimingInfo = {
+    requestTime: float,
+    redirectStart: float,
+    redirectEnd: float,
+    fetchStart: float,
+    dnsStart: float,
+    dnsEnd: float,
+    connectStart: float,
+    connectEnd: float,
+    tlsStart: float,
+    <!-- tlsEnd: float this should be the same as connectEnd -->
+    requestStart: float,
+    responseStart: float,
+    <!-- TODO responseHeadersEnd: float: Not sure quite what to use for this -->
+    responseEnd: float,
+};
+</pre>
+
+The <code>network.FetchTimingInfo</code> type represents the time of each part of
+the request, relative to <code>requestTime</code>.
+
+<div algorithm>
+To <dfn>get the fetch timings</dfn> given |request|:
+
+1. Let |global| be |request|'s [=request/client=].
+
+1. If |global| is null, return a map matching the
+   <code>network.FetchTimingInfo</code> production, with all fields set to 0.
+
+1. Let |timings| be |request|'s [=fetch timing info=].
+
+1. Let |connection timing| be |timings|' [=final connection timing info=] if
+   it's not null, or a new [=connection timing info=] otherwise.
+
+1. Let |request time| be [=convert fetch timestamp=] given [=convert fetch
+   timestamp=] given |timings|' <a spec=fetch>start time</a> and |global|.
```

Comment:
Looking at https://www.w3.org/TR/navigation-timing-2/#processing-model I feel like `start time` is only set for navigation requests (and preceeds the unload event). Is this something we should omit for non-navigation requests?

Reply:
https://w3c.github.io/resource-timing/ applies to all requests.